grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java

```diff
@@ -502,10 +502,22 @@ public void transportReportStatus(Status status, Metadata unused) {
     private void closeListener(Status newStatus) {
         if (!listenerClosed) {
             listenerClosed = true;
+
             ctx.logBuilder().responseContent(GrpcLogUtil.rpcResponse(newStatus, firstResponse), null);
+
             final boolean ok = newStatus.isOk();
-            setClientStreamClosed(ok);
-            messageFramer.close();
+            if (!clientStreamClosed) {
+                clientStreamClosed = true;
+                if (ok) {
+                    requestDeframer.close();
```

Comment:
Any idea to use an other way? This is the last reference to `HttpDeframer.close()`.

Reply:
I think it is safe to call `HttpDeframer.abort()` here